src/CMakeLists.txt

```diff
@@ -128,12 +128,20 @@ IF(GIT_FOUND AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/.git")
 	                OUTPUT_VARIABLE VERSION_STRING
 	                WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
 	)
-	string(STRIP "${VERSION_STRING}" VERSION_STRING) # Newline
+	string(STRIP "0.1~git${VERSION_STRING}" VERSION_STRING) # Newline
+ELSEIF(IS_DIRECTORY "${CMAKE_SOURCE_DIR}/.bzr")
+	execute_process(COMMAND bzr revno
+	                OUTPUT_VARIABLE VERSION_STRING
+	                WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+	)
+	string(STRIP "0.1~brz${VERSION_STRING}" VERSION_STRING) # Newline
 ELSE()
 	# Only do this if you really want to
 	IF(OVERRIDE_VCS)
 		message(WARNING "WARNING: No VCS checkout detected but overridden by user")
-		set(VERSION_STRING "0.0.0-norev")
+		if (NOT DEFINED VERSION_STRING)
+			set(VERSION_STRING "0.0.0-norev")
+		ENDIF()
 	ELSE()
 		message(FATAL_ERROR "No VCS checkout detected. Enable the OVERRIDE_VCS option if this is intended. This is not recommended.")
```

Comment:
Hmm, yeah.

I am not sure this is such a big thing to abort building. Also, what's the value of a version string if it's trivial to state any version you want anyway?

Older VCSes supported `$Id$` where you could insert a source file version string in the code, but this practice has been abandoned by `git`. As such, finding what was actually used to build a binary is near impossible nowadays.

Saying it in a different way, is it bad if we would just print a warning "he, we're missing a VCS checkout, and use 'unknown' as version", and then just build it?

We'd be missing the users that have a compiler installed and know about cmake and make and stuff, but don't pull the source from a repository. Seems to me that is a very small fraction of users. Sending source code around by mail isn't common any more.

what do you think?